cfme/tests/automate/test_domain.py

```diff
@@ -79,16 +80,56 @@ def test_domain_lock_disabled(request, appliance):
         initialEstimate: 1/16h
         tags: automate
     """
-    domain = appliance.collections.domains.create(
-        name=fauxfactory.gen_alpha(),
-        description=fauxfactory.gen_alpha(),
-        enabled=False)
-    request.addfinalizer(domain.delete_if_exists)
-    domain.lock()
-    view = navigate_to(domain, 'Details')
+    # Disable automate domain
+    with update(klass.namespace.domain):
+        klass.namespace.domain.enabled = False
+
+    # Adding schema for executing automate method
+    klass.schema.add_fields({'name': 'execute', 'type': 'Method', 'data_type': 'String'})
+
+    # Adding automate method
+    method = klass.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline'
+    )
+
+    # Adding instance to call automate method
+    instance = klass.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields={'execute': {'value': method.name}}
+    )
+
+    result = LogValidator(
+        "/var/www/miq/vmdb/log/automation.log",
+        matched_patterns=[r".*ERROR.*"],
+    )
+    result.fix_before_start()
+
+    # Executing automate method using simulation
+    simulate(
+        appliance=klass.appliance,
+        attributes_values={
+            "namespace": klass.namespace.name,
+            "class": klass.name,
+            "instance": instance.name,
+        },
+        message="create",
+        request="Call_Instance",
+        execute_methods=True,
+    )
+    result.validate_logs()
+
+    klass.namespace.domain.lock()
+    view = navigate_to(klass.namespace.domain, 'Details')
     assert 'Disabled' in view.title.text
     assert 'Locked' in view.title.text
 
+    # Need to unlock the domain to perform teardown on domain, namespace, class
```

Comment:
Where is the teardown happening btw ? 
Required : Delete the domain , namespace and class after test .

Reply:
Hi @sshveta,

- Actually  domain, namespace and class(i.e. klass) are [global fixtures](https://github.com/ManageIQ/integration_tests/blob/master/cfme/fixtures/automate.py#L5).
- But here domain locking is comes under exception case and to delete ```domain``` we need to unlock it first. Hence added this step.

Reply:
@ganeshhubale Does `instance` need any teardown?

Reply:
No. Actually, Instances or methods are under class. So their teardown happens when we delete the class itself.